src/codelabs/async-await/index.md

```diff
@@ -0,0 +1,498 @@
+  frameborder="no"
+  height="550"
+  width="100%">
+</iframe>
+
+  {{ site.alert.info }}
+    If your tests pass, you can ignore [info-level messages.](/guides/language/analysis-options#customizing-analysis-rules)
+  {{ site.alert.end }}
+
+## Handling errors
+To handle errors in an `async` function, use try-catch: 
+
+```dart
+  try {
+    var order = await getUserOrder();
+    print('Awaiting user order...');
+  } catch (err) {
+    print('Caught error: $err');
+  }
+```
+Within an `async` function, you can write [try-catch clauses](/guides/language/language-tour#catch)
+the same way you would in synchronous code. 
+
+### Example: async and await with try-catch
+Run the following example to see how to handle an error from an 
+asynchronous function. Can you guess what the output will be before running it?
+<!-- [//]: https://gist.github.com/25ade03f0632878a9169209e3cd7bef2 -->
+<iframe 
+  src="https://dartpad.dartlang.org/experimental/embed-new-dart.html?id=25ade03f0632878a9169209e3cd7bef2"
+  style="border: 1px solid lightgrey;"
+  frameborder="no"
+  height="525"
+  width="100%" >
+</iframe>
+
+### Exercise: Practice handling errors
+
+The following exercise provides practice handling errors with asynchronous code
+using the approach described in the previous section. To simulate asynchronous
+operations, your code will call the following function, which is provided for you:
+
+|------------------+-----------------------------------+-------------|
+| Function         | Type signature                    | Description |
+|------------------|-----------------------------------|-------------|
+| getNewUsername() | `Future<String> getNewUsername()` |      Returns the username from a user-initiated request for update|
+{:.table .table-striped}
+
+Use `async` and `await` to do the following:
+* Implement an asynchronous `changeUsername()` function that calls the provided
+ asynchronous function `getNewUsername()` and returns its result.
+  * Example return value from `changeUsername()`: `"jane_smith_92"` 
+  * Because `getNewUsername()` can encounter errors, the `changeUsername()` 
+  function must __catch and return any errors__ and it must __stringify the error before returning it__.
+* You can use the [toString()]({{site.dart_api}}/stable/dart-core/ArgumentError/toString.html) method to stringify both [Exceptions]({{site.dart_api}}/stable/dart-core/Exception-class.html) and [Errors.]({{site.dart_api}}/stable/dart-core/Error-class.html)
+
+<!-- [//]: https://gist.github.com/858f71f0ad0e70051999bcafa41806a3 -->
+
+<iframe 
+src="https://dartpad.dartlang.org/experimental/embed-new-dart.html?id=858f71f0ad0e70051999bcafa41806a3&theme=dark"
+frameborder="no"
+height="525"
+width="100%" >
+</iframe>
+
+### Exercise: Putting it all together
+
+It's time to practice what you've learned in one final exercise. 
+To simulate asynchronous operations, this exercise provides the asynchronous 
```

Comment:
Delete this sentence and the next one, or combine them somehow with the next paragraph. (As is, it's too repetitive.)